### PR TITLE
added snap manifest & workflow

### PR DESCRIPTION
**Pull request by soumyaDghosh**

### CODE REVIEW

File: .github/workflows/build-snap.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Snap
+on:
```
Comment (nlogozzo) [inline .github/workflows/build-snap.yml:2]: Could you make this:

```yml
on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]
    types: [ "review_requested", "ready_for_review" ]
  workflow_dispatch:
```

File: .github/workflows/build-snap.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    branches:
+      - main
+    tags-ignore:
+      - '**'
+permissions:
+  id-token: write
+  contents: read
+jobs:
+  build-snap:
+    runs-on: ubuntu-latest
```
Comment (nlogozzo) [inline .github/workflows/build-snap.yml:13]: add:
`name: "Gnome on Snap"`

File: .github/workflows/build-snap.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: Snap
```
Comment (nlogozzo) [inline .github/workflows/build-snap.yml:1]: remove line

File: .github/workflows/build-snap.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Checkout Git repository
+        uses: actions/checkout@v4
+      - name: Build Test Application Snap
+        uses: snapcore/action-build@v1
+        id: snapcraft-build
+        with:
+          snapcraft-args: "-v"
+      - uses: actions/upload-artifact@v4
+        with:
+          name: application-snap
```
Comment (nlogozzo) [inline .github/workflows/build-snap.yml:24]: `org.nickvision.application-snap`

**Review by nlogozzo — CHANGES_REQUESTED** [review]
Could you also rename the workflow file to just `snap.yml`?

File: snap/snapcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+      - -DCMAKE_INSTALL_PREFIX=/snap/application/current/usr
+      - -DCMAKE_BUILD_TYPE=Release
+    stage-packages:
+      - libstdc++6
+    build-environment:
+      - CMAKE_PREFIX_PATH: ${CRAFT_STAGE}/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}/cmake${CMAKE_PREFIX_PATH:+:$CMAKE_PREFIX_PATH}
+    parse-info: [usr/share/metainfo/org.nickvision.application.metainfo.xml]
+    override-prime: |
+      craftctl default
+      chmod 777 usr/lib/org.nickvision.application/org.nickvision.application.gnome
```
Comment (nlogozzo) [inline snap/snapcraft.yaml:85]: is this still needed, thought I fixed it?

**Review by nlogozzo — CHANGES_REQUESTED** [review]

File: .github/workflows/build-snap.yml (hunk 4 above)
Comment (soumyaDghosh) [reply .github/workflows/build-snap.yml:24]: This thing I think should be kept.. Snaps anyways doesn't use app_ids, and have `org.nickvision.application.snap` as name.. Instead it names `application_{version}_{arch}.snap`

Comment (nlogozzo) [reply .github/workflows/build-snap.yml:24]: ah ok makes sense then

File: .gitignore [hunk 6]
```diff
@@ -51,6 +51,8 @@ bin/
 .flatpak-builder/
 .flatpak/
 _nickbuild/
+*.snap
+*.txt
```
Comment (nlogozzo) [inline .gitignore:55]: *.txt files should not be ignored...

File: .github/workflows/snap.yaml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+    types: [ "review_requested", "ready_for_review" ]
+  workflow_dispatch:
+
```
Comment (nlogozzo) [inline .github/workflows/snap.yaml:8]: remove blank line

File: .github/workflows/snap.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+name: Snap
```
Comment (nlogozzo) [inline .github/workflows/snap.yaml:9]: remove line
having the "GNOME on Snap" line below is enough to name the workflow

File: .github/workflows/snap.yaml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+permissions:
+  id-token: write
+  contents: read
+jobs:
+  gnome-snap:
+    name: "Gnome on Snap"
```
Comment (nlogozzo) [inline .github/workflows/snap.yaml:15]: GNOME in all caps please

**Review by nlogozzo — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 6 above)
Comment (soumyaDghosh) [reply .gitignore:55]: Snap remote builds have txt files as the log file. So, what about `snap*.txt`?

File: .github/workflows/snap.yaml (hunk 8 above)
Comment (soumyaDghosh) [reply .github/workflows/snap.yaml:9]: Made it same as the flatpak one :thinking:

File: .gitignore (hunk 6 above)
Comment (nlogozzo) [reply .gitignore]: those files will never be committed to the repo...and someone using git wouldn't add them...and if they did we would reject them.

so no need for *.txt in the gitignore...

CMake files end in .txt, so if you put that here, changes to the CMake files wouldn't be read anymore lol

Comment (nlogozzo) [reply .gitignore]: the `.gitignore` applies to committed and watched changes for a repo

CI builds never commit to the main repo, so no need for anything here

Comment (soumyaDghosh) [reply .gitignore]: Uhh.. I made it application*txt

File: .github/workflows/build-snap.yml (hunk 3 above)
Comment (soumyaDghosh) [reply .github/workflows/build-snap.yml]: The name is not showing anymore

File: .gitignore (hunk 6 above)
Comment (nlogozzo) [reply .gitignore]: that doesn't make it any better IMO...

no need for the line at all, just because txt files are so commonly used already in this repo

Comment (soumyaDghosh) [reply .gitignore]: It just helped me... umm okay.. no pushing it

**Review by nlogozzo — APPROVED** [review]

**Pull request merged**